src/test/scala/nl.knaw.dans.easy.stage/dataset/AmdSpec.scala

```diff
@@ -0,0 +1,107 @@
+
+import scala.util.{ Success, Try }
+import scala.xml.PrettyPrinter
+
+class AmdSpec extends MdFixture {
+  // an indentation of zero allows simple comparison
+  private val prettyPrinter: PrettyPrinter = new scala.xml.PrettyPrinter(1024, 0)
+  private val nowIso = DateTime.now.toString(ISODateTimeFormat.dateTime())
+
+  "apply" should "validate for each test bag" in {
+    val depositorInfoDir: Path = sdoSetDir.toPath.resolve("metadata/depositor-info")
+    //assume(isAvailable(triedSchema))
+    for (bag <- new File("src/test/resources/dataset-bags").listFiles()) {
+      sdoSetDir.mkdirs()
+      val amd = AMD("foo", DateTime.now, "SUBMITTED", DepositorInfo(depositorInfoDir))
+      // unMarshall(amd) shouldBe a[Success[_]]
+      deleteDirectory(sdoSetDir)
+    }
+  }
+
+  it should "generate a remark" in {
+    val msg1 = "this dataset DOES NOT contain Privacy Sensitive data."
+    val msg2 = "Please contact me about blabla"
+    val info = DepositorInfo(acceptedLicense = Some(false), privacySensitiveRemark = msg1, messageFromDepositor = msg2)
+    val amd = AMD("foo", DateTime.now, "SUBMITTED", info)
+
+    val formattedAmd = prettyPrinter.format(amd)
+    formattedAmd should include(prettyPrinter.format(
+       <stateChangeDates>
+         <damd:stateChangeDate>
+           <fromState>DRAFT</fromState>
+           <toState>SUBMITTED</toState>
+           <changeDate>{ nowIso }</changeDate>
+         </damd:stateChangeDate>
+       </stateChangeDates>
+    ))
+
+    formattedAmd should include(prettyPrinter.format(
+      <remarks>
+        <remark>
+          <text>{ msg1 } { msg2 }</text>
+          <remarkerId>foo</remarkerId>
+          <remarkDate>{ nowIso }</remarkDate>
+        </remark>
+      </remarks>
+    ))
+    // not formatted remark (as shown in the webui):
+    (amd \\ "remarks" \\ "text").text shouldBe
+      s"""$msg1
+         |
+         |$msg2
+         |""".stripMargin.trim
+    // unMarshall(amd) shouldBe a[Success[_]]
+  }
+
+  it should "not generate state changes for a DRAFT (and an empty remark)" in {
+    val info = DepositorInfo(acceptedLicense = None, privacySensitiveRemark = "", messageFromDepositor = "")
+    val amd = AMD("foo", DateTime.now, "DRAFT", info)
+    val formattedAmd = prettyPrinter.format(amd)
+    formattedAmd should include(prettyPrinter.format(<stateChangeDates/>))
+    formattedAmd should include(prettyPrinter.format(<remarks></remarks>))
+  }
+
+  it should "unMarshall a published AMD" in {
+    val info = DepositorInfo(acceptedLicense = None, privacySensitiveRemark = "", messageFromDepositor = "")
+    val amd = AMD("IPDBSTest", DateTime.now, "PUBLISHED", info)
+
+    val file = new File(testDir + "/tested-amd.xml")
+    FileUtils.write(file, prettyPrinter.format(amd))
+    unMarshall(file) shouldBe a[Success[_]]
+  }
+
+  private def unMarshall(file: File) = {
+    Try {
+      JiBXObjectFactory.unmarshal( // how the webui reads it
+        new AdministrativeMetadataImpl().getClass,
```

Comment:
```suggestion
        classOf[AdministrativeMetadataImpl],
```

might give you the same without creating a new instance of the class